corehq/motech/dhis2/repeaters.py

```diff
@@ -30,6 +33,7 @@ class Meta(object):
     payload_generator_classes = (FormRepeaterJsonPayloadGenerator,)
 
     dhis2_config = SchemaProperty(Dhis2Config)
+    dhis2_version = StringProperty(default=None)
```

Comment:
Looks like this is being stored on a couch doc - will it need to be updated?  Is there some scenario where the upstream server changes version, but this property isn't auto-updated to reflect that?

Reply:
That is an excellent question!

Yes, it is conceivable that the DHIS2 server will be upgraded. 

The API version will only become a real problem when the new version is 4 or more versions ahead of the original version, in which case DHIS2 will return 404 errors on API calls. But it would be nice if it never got to that point.

Fetching metadata to get the DHIS2 version number is expensive. (I don't have an exact number, but when I tested just now it took between 14 and 15 "mississippi"s.)

1. We could store when last we checked `dhis2_version`, and if it's more than a year ago, check again. 
2. We could check on unexpected 404s. 
3. We could add a button to the Edit DHIS2 Repeater form that allows allows a user to check.

I'd prefer to leave the UI, if possible. Maybe (1) and (2)? Just (1) should be enough.

Reply:
Either of those seems reasonable to me - whatever you think is appropriate.